crates/c-api/include/wasmtime/config.h

```diff
@@ -367,6 +367,36 @@ WASM_API_EXTERN void wasmtime_config_cranelift_flag_enable(wasm_config_t*, const
  */
 WASM_API_EXTERN void wasmtime_config_cranelift_flag_set(wasm_config_t*, const char *key, const char *value);
 
+/**
+ * \brief Whether or not to enable support for asynchronous functions in Wasmtime.
+ *
+ * When enabled, the config can optionally define host functions with async. 
+ * Instances created and functions called with this Config must be called through their asynchronous APIs, however.
+ * For example using wasmtime_func_call will panic when used with this config.
+ *
+ * For more information see the Rust documentation at
+ * https://docs.wasmtime.dev/api/wasmtime/struct.Config.html#method.async_support
+ */
+WASMTIME_CONFIG_PROP(void, async_support, bool)
+
+/**
+ * \brief Configures the size of the stacks used for asynchronous execution.
+ *
+ * This setting configures the size of the stacks that are allocated for asynchronous execution. 
+ *
+ * The value cannot be less than max_wasm_stack.
+ *
+ * The amount of stack space guaranteed for host functions is async_stack_size - max_wasm_stack, so take care
+ * not to set these two values close to one another; doing so may cause host functions to overflow the stack 
+ * and abort the process.
+ *
+ * By default this option is 2 MiB.
+ *
+ * For more information see the Rust documentation at
+ * https://docs.wasmtime.dev/api/wasmtime/struct.Config.html#method.async_stack_size
+ */
+WASMTIME_CONFIG_PROP(void, async_stack_size, uint64_t)
+
```

Comment:
From an organizational perspective, what do you think about putting all of these functions in a new `wasmtime/async.h` header? Ideally we'd go as far as generating `wasmtime/__config__.h` or something which has `#define WASMTIME_ASYNC 0` in non-async builds of the C API, but we don't need to get that fancy yet. In the meantime though having what is almost a conditional header might be a good way to mitigate that in the meantime.

Reply:
And additionally I think it might be good to place all the Rust code in a `crates/c-api/src/async.rs` or similar.

Reply:
To be clear, you're talking about moving the function and linker methods into this new header too?

Reply:
Indeed yeah I think that'd be a good way to organize, although if you feel differently I don't feel strongly so it's ok to leave as-is too

Reply:
I actually wanted to do this at first so happy to update :)